### PR TITLE
Remove trailing whitespaces from formatted queries in some cases

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Remove trailing whitespaces from formatted queries in `clickhouse-client` or `clickhouse-format` in some cases.

Detailed description / Documentation draft:
Trailing whitespaces in queries copy-pasted from terminal look ugly.

### CODE REVIEW

File: src/Parsers/ASTExpressionList.cpp [hunk 1]
```diff
@@ -37,10 +37,8 @@ void ASTExpressionList::formatImplMultiline(const FormatSettings & settings, For
         {
             if (separator)
                 settings.ostr << separator;
-            settings.ostr << ' ';
```
Comment (azat) [inline src/Parsers/ASTExpressionList.cpp:40]: FWIW `TSVRaw` uses `formatImpl` and it will still has trailing whitespaces

Comment (alexey-milovidov) [reply src/Parsers/ASTExpressionList.cpp:40]: Ok, I will fix it also.

File: tests/queries/0_stateless/01069_database_memory.reference [hunk 2]
```diff
@@ -5,4 +5,4 @@ CREATE DATABASE memory_01069\nENGINE = Memory()
 4
 3
 4
-CREATE TABLE memory_01069.file\n(\n    `n` UInt8\n)\nENGINE = File(\'CSV\')
+CREATE TABLE memory_01069.file\n(`n` UInt8\n)\nENGINE = File(\'CSV\')
```
Comment (KochetovNicolai) [inline tests/queries/0_stateless/01069_database_memory.reference:8]: I suppose we need to remove `\n` before `(` and after `)`. Or restore previous behaviour.

File: tests/queries/0_stateless/01083_expressions_in_engine_arguments.reference [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 CREATE TABLE test_01083.rich_syntax AS remote(\'localhos{x|y|t}\', cluster(\'test_shard_localhost\', remote(\'127.0.0.{1..4}\', \'test_01083\', \'view\')))
-CREATE VIEW test_01083.view\n(\n    `n` Int64\n) AS\nSELECT toInt64(n) AS n\nFROM \n(\n    SELECT toString(n) AS n\n    FROM test_01083.merge\n    WHERE _table != \'qwerty\'\n    ORDER BY _table ASC\n)\nUNION ALL\nSELECT *\nFROM test_01083.file
-CREATE DICTIONARY test_01083.dict\n(\n    `n` UInt64, \n    `col` String DEFAULT \'42\'\n)\nPRIMARY KEY n\nSOURCE(CLICKHOUSE(HOST \'localhost\' PORT 9440 SECURE 1 USER \'default\' TABLE \'url\' DB \'test_01083\'))\nLIFETIME(MIN 0 MAX 1)\nLAYOUT(CACHE(SIZE_IN_CELLS 1))
+CREATE VIEW test_01083.view\n(`n` Int64\n) AS\nSELECT toInt64(n) AS n\nFROM \n(\n    SELECT toString(n) AS n\n    FROM test_01083.merge\n    WHERE _table != \'qwerty\'\n    ORDER BY _table ASC\n)\nUNION ALL\nSELECT *\nFROM test_01083.file
+CREATE DICTIONARY test_01083.dict\n(\n    \n    `n` UInt64,\n    \n    `col` String DEFAULT \'42\'\n)\nPRIMARY KEY n\nSOURCE(CLICKHOUSE(HOST \'localhost\' PORT 9440 SECURE 1 USER \'default\' TABLE \'url\' DB \'test_01083\'))\nLIFETIME(MIN 0 MAX 1)\nLAYOUT(CACHE(SIZE_IN_CELLS 1))
```
[the remark this answers is not in the document]
Comment (alexey-milovidov) [reply tests/queries/0_stateless/01083_expressions_in_engine_arguments.reference]: Fixed.

File: tests/queries/0_stateless/01069_database_memory.reference (hunk 2 above)
Comment (alexey-milovidov) [reply tests/queries/0_stateless/01069_database_memory.reference]: Now you can see that the test results only differ in trailing whitespaces, as expected.

**Pull request merged**